File: Algorithms/Greedy/Beautiful Pairs.py
import os

#
# Complete the 'beautifulPairs' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY A
#  2. INTEGER_ARRAY B
#

def beautifulPairs(A, B):
    # Write your code here
    count = 0
    for i, a in enumerate(A):
        for j, b in enumerate(B):
            if a == b:
                count += 1
                B[j] = -1
                break
    return count + 1 if count != n else count - 1


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    A = list(map(int, input().rstrip().split()))
    B = list(map(int, input().rstrip().split()))

    result = beautifulPairs(A, B)

    fptr.write(str(result) + '\n')
    fptr.close()

File: Algorithms/Greedy/Beautiful Pairs.php
<?php

/*
 * Complete the 'beautifulPairs' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER_ARRAY A
 *  2. INTEGER_ARRAY B
 */

function beautifulPairs($A, $B) {
    # Write your code here
    $countA = array_count_values($A);
    $countB = array_count_values($B);

    $matches = 0;

    foreach ($countA as $num => $count) {
        if (isset($countB[$num])) {
            $matches += min($count, $countB[$num]);
        }
    }

    if ($matches == count($A)) {
        return $matches - 1;
    }
    else {
        return $matches + 1;
    }
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$A_temp = rtrim(fgets(STDIN));
$A = array_map('intval', preg_split('/ /', $A_temp, -1, PREG_SPLIT_NO_EMPTY));

$B_temp = rtrim(fgets(STDIN));
$B = array_map('intval', preg_split('/ /', $B_temp, -1, PREG_SPLIT_NO_EMPTY));

$result = beautifulPairs($A, $B);

fwrite($fptr, $result . "\n");
fclose($fptr);
